buildenv/jenkins/upstream_openjdk_tests

```diff
@@ -0,0 +1,306 @@
+    }
+}
+
+// parallel build job configs
+Map parallelJobs = [:]
+
+// Only schedule GH API work on x86 linux nodes as there
+// are plenty available.
+node("ci.role.test&&hw.arch.x86&&sw.os.linux") {
+    parameters {
+        string(name:'VERSIONS', defaultValue: '8 11', description: "Space separated list of versions to check")
+        string(name:'GH_USERNAME', description: "Github API username to authenticate as")
+        string(name:'GH_TOKEN', description: "Github API OAuth personalized access token to be used for authentication")
+        booleanParam(name:'INCLUDE_EA', defaultValue: true, description: "Whether or not to include EA builds")
+        booleanParam(name:'FORCE_RETEST', defaultValue: false, description: "Whether or not to force a re-test irrespective previous runs")
+    }
+
+    // All params have sensible defaults
+    List versions = params.VERSIONS ? params.VERSIONS.split(" ") : "8 11".split(" ")
+    boolean includeEA = params.INCLUDE_EA ? params.INCLUDE_EA : true
+    boolean forceRetest = params.FORCE_RETEST ? params.FORCE_RETEST : false
+    String user = params.GH_USERNAME ? params.GH_USERNAME : "gh-user-not-set"
+    String token = params.GH_TOKEN ? params.GH_TOKEN : "gh-token-not-set"
+    String userTokenBase64 = Base64.getEncoder().encodeToString((user + ":" + token).getBytes())
+    Map releaseCtxts = [:]
+    Map latestRelInfos = [:]
+
+    TEST_JOB_NAME = "${env.JOB_NAME}"
+    File workspace = new File("${env.WORKSPACE}")
+    workspace.mkdirs()
+    versions.each { version ->
+        ReleaseRetriever retriever = new ReleaseRetriever(version, includeEA, userTokenBase64)
+        List ctxts = retriever.getReleaseContexts()
+        latestRelInfos.put(version, new LatestReleaseInfo(workspace, version, retriever.getLatestReleaseDate()))
+        releaseCtxts.put(version, ctxts)
+    }
+    dir("$env.WORKSPACE") { // use workspace relative directory
+        versions.each { version ->
+            stage("prepare OpenJDK Version: $version") {
+                LatestReleaseInfo relInfo = latestRelInfos.get(version)
+                List ctxts = releaseCtxts.get(version)
+                ctxts.each { println it.toString() }
+                if (forceRetest || relInfo.isRetestNeeded()) {
+                    // trigger tests for each context
+                    ctxts.each{ r -> 
+                        stage("prepare $r.OS-$r.ARCH") {
+                            def jdk_url = r.JDK_URL
+                            def source_url = r.SOURCE_URL
+                            def jdk_version = version
+                            def jenkins_file = r.getJenkinsFile()
+                            def jdk_jre_url = "$r.JDK_URL $r.JRE_URL"
+                            parallelJobs["Test_openjdk${version}_hs_sanity.openjdk_${r.ARCH}_${r.OS}"] = {
+                                // Run a _sanity.openjdk test job with appropriate parameters
+                                build job: TEST_JOB_NAME, parameters: [
```

Comment:
This 'TEST_JOB_NAME' is exactly same as the branch name.

Reply:
My understanding was that the key to the `parallelJobs` map is the "branch name". At least jenkins calls it that when it runs. "TEST_JOB_NAME" refers to the jenkins job name, no? Like "Grinder". Scheduling jobs that way assumes existing job config (with matching name referenced via `TEST_JOB_NAME`), does it not?

Reply:
Branch name is not necessarily same as the test job name. It's nice to include the hint about the test jobs. For this case as long as the branch name including the information as '${version}, hotspot, sanity, arch, os ' it would be enough. So we know which test job will run.  

Yes, TEST_JOB_NAME refers to the jenkins job name, which is the test jobs we have generated. for example: job https://ci.adoptopenjdk.net/view/Test_upstream/job/Test_upstream_openjdk8_hs_sanity.openjdk_x86-64_linux/

For now we only have one job generated, we can either generate others by duplicate copy or by autogen. If we don't want the job failed by non-generated jobs we could also use @Library('openjdk-jenkins-helper@master') by check if (JobHelper.jobIsRunnable("${TEST_JOB_NAME}")) { build job }

Reply:
Hmm, I see you prefer to create a new job and all parallel test jobs are also clone of this new job?
Say the job number is 1, and then you have 6 test jobs. Those jobs will be job number 2,3,...,7.

Reply:
We have tried this strategy before for parallel jobs and think it's not clear/straightforward for triage as all test jobs seem similar. You can only go to console and know jobs targets. 
<img width="338" alt="Screen Shot 2019-06-26 at 9 49 39 AM" src="https://user-images.githubusercontent.com/18030625/60185451-f682e300-97f7-11e9-9b06-c0d154dd56e1.png">

See this picture it's not clear to know which job is running against 11|8. 

My original thought is to create a new job and this new job will trigger all jobs you want. Similarly to https://ci.adoptopenjdk.net/view/Test_external_weekly/job/openjdk_externaltest_extended_weekly/configure

But I'm ok with both.

Reply:
@sophia-guo It makes sense what you are saying. I'll implement it so that the generated jobs have a good name as you suggest for discoverability. What I was trying to point out is that there seems to be a "branch name" and "job name" concept in jenkins (they're not necessarily the same). I understand the branch and job names should match for a revised version of this patch. It doesn't currently, which I'll fix. Thanks and sorry for the confusion.